test/puppetlabs/trapperkeeper/services/metrics/metrics_service_test.clj

```diff
@@ -58,7 +69,24 @@
          (is (= 200 (:status resp)))
          (doseq [[metric path] body
                  :let [resp (http-client/get (str "http://localhost:8180/metrics/v1" path))]]
-           (is (= 200 (:status resp))))))
+           (is (= 200 (:status resp)))))
+
+       (let [resp (http-client/get "http://localhost:8180/metrics/v2/list")
+             body (parse-response resp)]
+         (is (= 200 (:status  resp)))
+         (doseq [[namesp mbeans] (get body "value") mbean (keys mbeans)
+                 :let [url (str "http://localhost:8180/metrics/v2/read/"
+                                (jolokia-encode (str namesp ":" mbean))
+                                ;; NOTE: Some memory pools intentionally don't
+                                ;; implement MBean attributes. This results
+                                ;; in an error being thrown when those
+                                ;; attributes are read and is expected.
+                                "?ignoreErrors=true")
```

Comment:
This is mostly an apples-to-apples implementation of the test for the v1 endpoint above which is to check that every metric name returned by a "list" operation can actually be retrieved by a "read" operation. I could re-factor that, but it will change the character of both tests.

The `ignoreErrors` parameter is a bit broadly named, basically it toggles between returning an error message thrown by attribute retrieval inline:

    "CollectionUsageThreshold": "javax.management.RuntimeMBeanException: java.lang.UnsupportedOperationException: CollectionUsage threshold is not supported"

vs. failing the entire request even if other MBean attributes were readable. The `java.lang:type=MemoryPool,name=Metaspace` MBean intentionally does not implement a couple of standard MemoryPool attributes, so `ignoreErrors` needs to be toggled to `true` if the readability of all MBeans is being checked.